scripts/ensure-frozen-migrations

```diff
@@ -24,39 +21,62 @@ show_diff() {
     diff -u <(remove_comments "$file1") <(remove_comments "$file2")
 }
 
+fail() {
+    echo "❌ Existing migrations changed. Please make sure only comments are changed in existing migration files."
+    exit 1
+}
+
 main() {
     local sql_dir="backend/controller/sql/schema"
     local changed_files
-    local merge_base
+    local all_old_files
+    local new_files=( )
 
-    # Find the merge base commit
-    merge_base=$(git merge-base HEAD origin/main)
+    head_sha=$(git merge-base HEAD origin/main)
+    changed_files=$(git diff --name-only "$head_sha" -- "$sql_dir"/*.sql)
+    all_old_files=$(git diff --name-only origin/main -- "$sql_dir")
```

Comment:
Why did this change? origin/main is almost never correct in CI because if you're behind main it will break completely.

Reply:
Ah good catch. That should be `$head_sha`. Will fix. It accidentally changed because I was rewriting a lot.

Reply:
Actually I'm not sure how I got it like this. Changed files should be essentially same as `all_old_files`. Maybe it was adding --name-only on one of them but didn't need the other. I will revisit.